### PR TITLE
Post: comments for posts! 

### PR DESCRIPTION
**Pull request by willybeans**
in reference to [this issue](https://github.com/7evam/pie-madness-api/issues/1)
This PR creates a new route that allows users to add comments to posts. 

it generates a UUID on request for the comment itself

NOTE: 
i think the only error case that is missing is this: `send an error message if the post the comment is being added to does not exist`
i believe id have to add a new route that getsPostsById to confirm they exist

### CODE REVIEW

**Comment by willybeans** [comment]
@7evam do you want an error thrown if the timestamp is missing?

File: api/routes/comments/comments.controller.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  const { postId } = req.params;
+  const { body } = req;
+
+  const validationErrors = validateRequiredPostFields(
+    {
+      postId: {
+        required: true,
+        notEmpty: true,
+      },
+      text: {
```
Comment (7evam) [inline api/routes/comments/comments.controller.js]: should be comment, not text

File: api/routes/comments/comments.service.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+const { dynamodb, TableName } = require("../../services/db");
+
+exports.createComment = async (postId, commentId, params) => {
+  const newComment = {
+    TableName,
+    Item: {
+      PK: `POST#${postId}`,
+      SK: `COMMENT#${commentId}`,
+      text: params.text,
```
Comment (7evam) [inline api/routes/comments/comments.service.js]: should be params.comment, not params.text

File: api/routes/comments/comments.controller.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+  if (validationErrors.error) {
+    res.status(400).json(validationErrors);
+    return;
+  }
+
+  const isValid = await authenticateUser(body.userId, body.secret);
+
+  if (isValid) {
+    const uuid = crypto.randomUUID();
```
Comment (7evam) [inline api/routes/comments/comments.controller.js]: please create the uuid in the service. i would consider this business logic so it belongs in service. controller validates and handles request/response only

**Review by 7evam — CHANGES_REQUESTED** [review]
looks good! tried this locally and it worked, just added some minor changes before merge

File: api/routes/comments/comments.controller.js (hunk 1 above)
Comment (willybeans) [reply api/routes/comments/comments.controller.js]: 👍

File: api/routes/comments/comments.service.js (hunk 2 above)
Comment (willybeans) [reply api/routes/comments/comments.service.js]: 👍

**Comment by willybeans** [comment]
@7evam ready for review + merge

**Review by 7evam — APPROVED** [review]
looks good, thanks for fixes!

**Pull request merged**